examples/apt_blueprint/blueprint.py

```diff
@@ -54,6 +56,49 @@
 )
 
 
+@aptb.action_enumerate
+def action_enumeration(auth: AuthState, params: Dict[str, Set]) -> List[ActionStatus]:
+    """
+    This is an optional endpoint, useful for allowing requestors to enumerate
+    actions filtered by ActionStatus and role.
+
+    The params argument will always be a dict containing the incoming request's
+    validated query arguments. There will be two keys, 'statuses' and 'roles',
+    where each maps to a set containing the filter values for the key. A typical
+    params object will look like:
+
+        {
+            "statuses": {<ActionStatusValue.ACTIVE: 3>},
+            "roles": {"creator_id"}
+        }
+
+    Notice that the value for the "statuses" key is an Enum value.
+    """
+    statuses = params["statuses"]
+    roles = params["roles"]
+    matches = []
+
+    for _, action in simple_backend.items():
```

Comment:
Unfortunately the entire implementation is specific to this example and pretty unrealistic (since it's using a dict as the data store). As I was thinking through implementing enumeration for the example, I realized iterating through every action would, in practice, be rough. So this endpoint would require smart use of indexes/filters to offload as much work as possible from python to the storage backend.

Given how different backend implementations can be, all I can think of is defining an abstract base class and asking users to implement their own subclasses. We could use some of the `dynamo_helpers` + experiences using the helpers to define an ABC interface. I'm also open to discussing other ideas